src/pages/_app.js

```diff
@@ -1,27 +1,34 @@
 import { useState } from "react";
 import { useRouter } from "next/router";
-import transactionsData from "@/src/data/transactions.json";
+import transactions from "@/src/data/transactions.json";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
-  const [transactions, setTransactions] = useState(transactionsData);
+  const [initialData, setInitialData] = useState(transactions);
   const [mode, setMode] = useState("default");
   const [action, setAction] = useState("default");
   const router = useRouter();
 
   function handleMode(mode) {
     setMode(mode);
   }
+
+  function handleAddTransaction(data) {
+    setInitialData([{ id: uid(), ...data, amount: parseFloat(data.amount) }, ...initialData]);
+  }
+
   function handleCancel() {
     handleMode("default");
+    router.push("/");
```

Comment:
Since line 22 is unnecessary, you don't need this function. Instead, directly pass your handleMode function as onMode and use onClick={() => onMode("default")}.